Karma-Khameleons/bangularMobile
# Sam product pages

## Description
This allows users to see a list of ALL products, with hyperlinks to detail views of each product on that list. 


## Number of Fixes
1. Users can now see a list of all products
1. Users can now see the product details for any product

## Related Ticket(s)
#2 


## Problem to Solve
Users couldn't browse products, and http requests were being made using hard-coded resource urls.

## Changes
Reformatted ProductFactory to use RootFactory instead of hard-coded resource urls.

## Expected Behavior
Product list and product detail pages should render with all appropriate details


## Steps to Test Solution
1. Run the following commands:
```
git fetch --all
git checkout origin/sam-ProductPages
```
2. Then, with your server running, visit ```http://localhost:8080/#!/product_list```
3. Check that all products in the database are listed on this page
4. Click a link to a product, and ensure that product title, description, price, and a "buy product" button are all rendered.

## Documentation
